mock/ecs_pod_creator.go

```diff
@@ -2,21 +2,40 @@ package mock
 
 import (
 	"context"
-	"errors"
 
 	"github.com/evergreen-ci/cocoa"
+	"github.com/pkg/errors"
 )
 
 // ECSPodCreator provides a mock implementation of a cocoa.ECSPodCreator
 // that produces mock ECS pods. It can also be mocked to produce a pre-defined
 // cocoa.ECSPod.
-type ECSPodCreator struct{}
+type ECSPodCreator struct {
+	cocoa.ECSPodCreator
+
+	CreatePodInput  *cocoa.ECSPodCreationOptions
+	CreatePodOutput *cocoa.ECSPod
+}
+
+// NewECSPodCreator creates a mock ECS Pod Creator backed by the given Pod Creator.
+func NewECSPodCreator(c cocoa.ECSPodCreator) *ECSPodCreator {
+	return &ECSPodCreator{
+		ECSPodCreator: c,
+	}
+}
 
 // CreatePod saves the input and returns a new mock pod. The mock output can be
 // customized. By default, it will create a new pod based on the input that is
 // backed by a mock ECSClient.
 func (m *ECSPodCreator) CreatePod(ctx context.Context, opts ...*cocoa.ECSPodCreationOptions) (cocoa.ECSPod, error) {
-	return nil, errors.New("TODO: implement")
+	merged := cocoa.MergeECSPodCreationOptions(opts...)
```

Comment:
You don't need to merge the options when the input will already be merged by the implementation in L38. I would instead store a slice of ECSPodCreationOptions as the input.